src/api.js

```diff
@@ -0,0 +1,94 @@
+import { Promise } from 'es6-promise';
+
+class API {
+  constructor(addon) {
+    this.addon = addon;
+  }
+
+  // Currencies
+  getCurrencies(query) {
+    let self = this;
+
+    return new Promise((resolve, reject) => {
+      self.addon.request({
+        method: 'GET',
+        endpoint: 'currencies',
+        query: query,
+      }).then(response => resolve(response)).catch((err) => reject(err));
+    });
+  }
+
+  // Institutions
+  getInstitutions(query) {
+    let self = this;
+
+    return new Promise((resolve, reject) => {
+      self.addon.request({
+        method: 'GET',
+        endpoint: 'institutions',
+        query,
+      }).then(response => resolve(response)).catch((err) => reject(err));
+    });
+  }
+
+  getInstitution(id) {
+    let self = this;
+
+    return new Promise((resolve, reject) => {
+      self.addon.request({
+        method: 'GET',
+        endpoint: `institutions/${id}`,
+      }).then(response => resolve(response)).catch((err) => reject(err));
+    });
+  }
+
+  pollInstitution(id, v) {
+    let self = this;
+
+    return new Promise((resolve, reject) => {
+      self.addon.request({
+        method: 'GET',
+        endpoint: `institutions/${id}/poll?v=${v}`,
+        query: query,
+      }).then(response => resolve(response)).catch((err) => reject(err));
+    });
+  }
+
+  syncInstitution(id) {
+    let self = this;
+
+    return new Promise((resolve, reject) => {
+      self.addon.request({
+        method: 'GET',
```

Comment:
Humm now this makes me wonder if the sync endpoint would be better as a PUT or a POST. Technically GET doesn't modify the target institution, however a sync does modify the target...

Reply:
@siboulet you're right this should be a POST like the /institutions/sync endpoint